internal/template/utilization_functions.go

```diff
@@ -72,69 +73,108 @@ scalar(
+	// however the dependencies of the Prometheus server conflict.
 
-	// Always include the trial namespace
-	labels := []string{fmt.Sprintf("namespace=\"%s\"", data.Trial.Namespace)}
+	// Allow the extra label matchers to be variadic or comma-delimited
+	extraLabelMatchers = strings.Split(strings.Join(extraLabelMatchers, ","), ",")
 
-	for _, label := range strings.Split(strings.Join(labelArgs, ","), ",") {
-		if label == "" {
+	// Always include the trial namespace first
+	labelMatchers := make([]string, 0, 1+len(extraLabelMatchers))
+	labelMatchers = append(labelMatchers, fmt.Sprintf("namespace=\"%s\"", metricData.Trial.Namespace))
+	for _, labelMatcher := range extraLabelMatchers {
+		if labelMatcher == "" {
 			continue
 		}
 
-		kvpair := strings.Split(label, "=")
-		if len(kvpair) != 2 {
-			return "", fmt.Errorf("invalid label for utilization query, expected key=value, got: %s", label)
+		key, op, value, err := splitLabelMatcher(labelMatcher)
+		if err != nil {
+			return "", err
 		}
 
-		// Sanitize label name for prometheus usage
-		kvpair[0] = invalidLabelCharRE.ReplaceAllString(kvpair[0], "_")
-
-		labels = append(labels, fmt.Sprintf("label_%s=\"%s\"", kvpair[0], kvpair[1]))
+		labelMatchers = append(labelMatchers, key+op+value)
 	}
 
+	// Add the metric selector start and end markers (we know it will be non-empty because of the namespace)
+	metricSelector := fmt.Sprintf("{%s}", strings.Join(labelMatchers, ","))
+
+	// Wrap the standard metric data with the additional MetricSelector
 	input := struct {
 		MetricData
-		Labels string
+		MetricSelector string
 	}{
-		data,
-		fmt.Sprintf("{%s}", strings.Join(labels, ",")),
+		MetricData:     metricData,
+		MetricSelector: metricSelector,
 	}
 
+	// Panic if the query in the source code does not parse
+	tmpl := template.Must(template.New("query").Parse(query))
+
+	// Execute the template
 	var output bytes.Buffer
 	if err := tmpl.Execute(&output, input); err != nil {
 		return "", err
 	}
 
 	return output.String(), nil
 }
+
+// splitLabelMatcher returns the label name, operator and value from a label matcher.
+func splitLabelMatcher(lm string) (string, string, string, error) {
+	kv := regexp.MustCompile(`(=|!=|=~|!~)`).Split(lm, 2)
+	if len(kv) != 2 {
+		return "", "", "", fmt.Errorf("invalid label matcher: %s", lm)
+	}
+
+	key := strings.TrimSpace(kv[0])
+	op := lm[len(kv[0]) : len(lm)-len(kv[1])]
+	value := strings.TrimSpace(kv[1])
+
+	// Allow quotes to be optional since specifying them in the Go template would make things complicated
+	if !strings.HasPrefix(value, `"`) {
+		value = strconv.Quote(value)
+	}
+
+	// TODO We should remove this
+	// In the case of our built-in Prometheus, we can just alter the scape configs to not include the prefix
+	// In the case of a managed Prometheus we won't have control over this prefix and it may interfere with metric selection
+	if !strings.HasPrefix(key, "label_") {
+		key = "label_" + key
+	}
```

Comment:
this specifically comes in for how kube state metrics handles the translation of the pod labels into prometheus labels. 
given where this comes into play and the dependency on kube state metrics, im not sure it's worth having the `TODO` around removing this.

Reply:
We can always relabel metrics however we want in the built-in Prometheus to strip the `label_` prefix ; and for a managed Prometheus we can't possibly know what relabel configs they have applied, so even if that's the convention there is no guarantee that is what they are using.